### ISSUE TITLE
Add option to show all recipes, regardless of hidden catalysts

### ISSUE THREAD

**Issue by codetaylor** [mezz/JustEnoughItems#1209]
### What

I would like the option to temporarily disable the feature introduced in commit 156f46ed2be24d7f9d36bf515ded4d776fd223d7: `Hide recipe categories when all their crafting catalysts are hidden`.

### Why

I think that this feature is a good addition to JEI and I support it, however, it has greatly impeded my workflow. I am working on a pack with more than fifty non-linear stages using mods such as GameStages, ItemStages, and RecipeStages. While working on this project, I have come to rely on JEI as the powerful recipe browsing tool it is. With the introduction of this new feature, I am no longer able to view all recipes without unlocking all stages first. I would love to be able to temporarily disable this feature, possibly with a config option, and once again have a complete recipe catalog at my fingertips during development.

**Comment by mezz** [comment]
Thanks for the report!
Are there any other things that you would need disabled? For instance if you are using RecipeStages you may have a similar problem. It's not clear to me why this particular feature would be in the way of your workflow but others are not, so it would be helpful if you could explain a bit.

**Comment by codetaylor** [comment]
I feel foolish. After writing a detailed analysis of my workflow and the structure of my project, I gained insight into my problem and it occurred to me that you are right. Any feature of any mod that hides items in JEI based on game stages will hinder my approach. Adding the option to ignore the visible catalyst check will not solve my problem because my problem is, in fact, my workflow. The technique I established to conquer this mountain of work will need to be rethought.

Thank you for your time and all you do for the community!
